### PR TITLE
Reviewed group babylon api solution :

### PR DESCRIPTION
**Pull request by johnfolly**
- deleted unused declaration in dry run
- updated dry run message
- added function parameter Optional[str] typing
- added delete via -i , file option
- fixed error while getting fields and parsing data in file code
- deleted Option non necessary args : type, required
- fixed click options and arguments var name format use a-name-of-arg instead of a_name_of_arg
- added return type to create and delete
- added missing deploy and platform key
- fixed typos

### CODE REVIEW

**Comment by johnfolly** [comment]
## Deploy.yaml channgelog

- Added :
   - `simulator_repository`: the solution simulator repository (repository value in Solution.yaml)

  - `simulator_version`: the solution simulator version (version value in Solution.yaml)

  - `simulator_url`: the solution simulator url (url value in Solution.yaml)

- Decraped:
  - `acr_image_reference`: ACR image reference (my_repository:my_tag)


```yaml
# ACR image reference (my_repository:my_tag)
acr_image_reference: ""
# acr_image_reference (repository:tag) deprecated : image repository should be simulator_repository and tag sould be simulator_version
# simulator_repository: the solution simulator repository (repository value in Solution.yaml)
simulator_repository: ""
# simulator_version: the solution simulator version (version value in Solution.yaml)
simulator_version: ""
# simulator_url:the solution simulator url (url value in Solution.yaml)
simulator_url: ""
```

File: Babylon/groups/api/groups/solution/commands/create.py [hunk 1]
```diff
@@ -123,7 +124,11 @@ def create(
     if select:
         env.configuration.set_deploy_var("solution_id", retrieved_solution["id"])
 
-    logger.info(f"Created new solution with id: {retrieved_solution['id']}")
+    logger.info("Created new solution with \n"
+                f" - id: {retrieved_solution['id']}\n"
+                f" - key: {retrieved_solution['key']}\n"
+                f" - repository: {retrieved_solution['repository']}\n"
+                f" - version: {retrieved_solution['version']}")
     logger.debug(pformat(retrieved_solution))
```
Comment (o-reo) [inline Babylon/groups/api/groups/solution/commands/create.py:132]: Looks like info and debug will log the same message, you can probably remove one.

Comment (o-reo) [inline Babylon/groups/api/groups/solution/commands/create.py:132]:
```suggestion
```

**Review by o-reo — APPROVED** [review]
:+1:

**Review by o-reo — APPROVED** [review]
:+1:

**Comment by o-reo** [comment]
:rocket:

**Pull request merged**